### ISSUE TITLE
perceval.errors.RepositoryError: [Errno 2] No such file or directory: 'git'

### ISSUE THREAD

**Issue by bvasiles** [grimoirelab/perceval#99]
Hi,

I have a `git` log file in `/tmp/gitlog.log` and I'm running `sudo docker run --rm -it --name perceval -v ~/.perceval/cache:/root/.perceval/cache grimoirelab/perceval:latest git /tmp/gitlog.log`.

It fails with the following trace:
```
[2016-12-13 16:42:36,117] - Sir Perceval is on his quest.
[2016-12-13 16:42:36,119] - Fetching commits: '/tmp/gitlog.log' git repository from 1970-01-01 00:00:00+00:00; all branches
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 782, in _exec
    cwd=cwd, env=env)
  File "/usr/local/lib/python3.4/subprocess.py", line 856, in __init__
    restore_signals, start_new_session)
  File "/usr/local/lib/python3.4/subprocess.py", line 1460, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'git'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 261, in run
    for commit in commits:
  File "/usr/local/lib/python3.4/site-packages/perceval/backend.py", line 192, in decorator
    for data in func(self, *args, **kwargs):
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 109, in fetch
    commits = self.__fetch_and_parse_log(from_date, branches)
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 122, in __fetch_and_parse_log
    repo = self.__create_and_update_git_repository()
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 128, in __create_and_update_git_repository
    repo = GitRepository.clone(self.uri, self.gitpath)
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 639, in clone
    cls._exec(cmd, env={'LANG' : 'C'})
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 785, in _exec
    raise RepositoryError(cause=str(e))
perceval.errors.RepositoryError: [Errno 2] No such file or directory: 'git'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/perceval", line 179, in <module>
    main()
  File "/usr/local/bin/perceval", line 100, in main
    cmd.run()
  File "/usr/local/lib/python3.4/site-packages/perceval/backends/core/git.py", line 268, in run
    raise RuntimeError(str(e))
RuntimeError: [Errno 2] No such file or directory: 'git'
```

**Comment by sduenas** [comment]
The way to run this backend with a log file is as follows:

```
perceval git --git-log <logpath> <uri>
```

For your case, you have to replace `<logpath>` by `/tmp/gitlog.git`, and `<uri>` with anything you want, for instance with the path to that Git repository or with its URL. This `<uri>` won't be used during the fetching process but to tag those items fetched from the log file.

**Comment by bvasiles** [comment]
Exactly the same error when I run `sudo docker run --rm -it --name perceval -v ~/.perceval/cache:/root/.perceval/cache grimoirelab/perceval:latest git --git-log /tmp/gitlog.log [path to git repo on disk]`

**Comment by acs** [comment]
@bvasiles I think that "/tmp/gitlog.log" does not exists inside the container. You should copy it to the container (docker cp) or map it with "-v" as you did for the perceval cache.

**Comment by bvasiles** [comment]
@acs Good suggestion! Unfortunately that doesn't solve it. It still fails with the same error when I run `sudo docker run --rm -it --name perceval -v ~/.perceval/cache:/root/.perceval/cache grimoirelab/perceval:latest git 'https://github.com/grimoirelab/perceval.git' --from-date '2016-01-01'`

**Comment by albertinisg** [comment]
@bvasiles the issue was simpler, git package was not available inside the container :see_no_evil:. It was used to install perceval and then purged to decrease the image size. I've just made a PR with the fix.

Also, as the `git` backend does not uses cache but `repositories`, I would suggest to map the whole `.perceval` directory instead of just the cache:


```
docker run --rm -it --name perceval -v ~/.perceval:/root/.perceval grimoirelab/perceval:latest git 'https://github.com/grimoirelab/perceval.git' --from-date '2016-01-01'
 ```

**Comment by sduenas** [comment]
@bvasiles it should be fixed with the latest version of the Docker image. You can build one on your own or wait for DockerHub build gets completed.

**Comment by bvasiles** [comment]
Thanks everyone! Should we update the documentation [here](https://github.com/grimoirelab/perceval/tree/master/docker/images)?

**Comment by sduenas** [comment]
@bvasiles you're totally right. @albertinisg can you update the information related to docker support?

**Comment by sduenas** [comment]
@albertinisg I've just merged your PR (#112) regarding the change on the volume path. In any case, you still have to update the documentation to update the contents of the package. I think it includes now Git.
